projects/plugins/jetpack/extensions/blocks/map/map.php

```diff
@@ -165,3 +165,31 @@ function render_single_block_page() {
 	exit;
 }
 add_action( 'wp', __NAMESPACE__ . '\render_single_block_page' );
+
+/**
+ * Helper function to generate the markup of the block in PHP.
+ *
+ * @param Array $points - Array containing geo location points.
+ *
+ * @return string Markup for the jetpack/map block.
+ */
+function map_block_from_geo_points( $points ) {
+	$map_block_data = array(
```

Comment:
I wonder if we should add a check for a valid API key here? On WordPress.com sites there will always be a valid Mapbox API key, but that won't be the case on Jetpack sites.

Reply:
Do you mean only create the block if there is a valid key?
If we don't create the block now, then it will be difficult to create it later

Reply:
Good point. I suppose we can stil create the block, and then the site owner will have to edit the post later on to add the key. That works too, I suppose 👍